### ISSUE TITLE
7.3.3(4/5) and 7.3.2(5/4) Wording for use of current instance.

### ISSUE THREAD

**Issue by ARG-Editor** [pat-rogers/Ada-202x-WG9-Informal-Review#177]
From nholsti in #149, paraphrased:

In 7.3.3(4/5) the phrase "a name that denotes the declaration" is referring to the current instance of the type. But the analogous definition of Type_Invariant in RM 7.3.2 (5/4) uses a different phrase: "the identifier of the first subtype of the associated type". Why is the wording different in these two places, and which is correct?

**Comment by ARG-Editor** [comment]
Nholsti pointed out that we hadn't addressed this question: 

Interesting discussions, but no-one has commented on the second part of my original issue, which is that the analogous definition of the current instance for Type_Invariant in RM 7.3.2 (5/4) uses a different phrase: "the identifier of the first subtype of the associated type", instead of "a name that denotes the declaration". The former suggests a direct comparison of identifier strings on the lexical level, as for the identifiers that often follow "end" keywords, while the latter suggests the outcome of name resolution. Is the wording intentionally different? Or would it be better to use the same wording in both places?

I haven't checked all the other similar rules in the RM.

**Comment by ARG-Editor** [comment]
This is a separate issue from #149, so I've split it to treat it separately.

The wording in question (7.3.2(5/4)) is marked Redundant, which means that it is supposed to follow from other rules and be purely informational. The other wording (7.3.3(4/5)) is not marked that, probably because it is tied up in the notional type NT wording.

Probably these should be worded similarly, and the 7.3.2(5/4) wording be changed. But it is not wrong, and it is original Ada 2012 wording (it predates the 8.6 wording that makes it redundant) -- which makes it out of bounds for this review. As noted elsewhere (see reply to #15), "improving" wording has a significant possibility of introducing errors where there are none, and wording that has 8+ years of history cannot be a major problem in understanding (as if it was we would have already heard about it), So this issue, like all wording improvements is marked Deferred (we will gather these up and look at them post Ada 202x).

**Comment by sttaft** [comment]
I agree with your conclusion -- the wording is inconsistent, but not misleading, and better not to try to "fix" it now when we know how frequently "benign" last-minute changes can create new problems.  But definitely something to revisit later.  Thanks, Niklas, for your very careful reading!
